src/WorkOS.net/Services/UserManagement/UserManagementService.cs

```diff
@@ -82,5 +82,24 @@ public UserManagementService(WorkOSClient client)
             };
             return await this.Client.MakeAPIRequest<(User, Session)>(request, cancellationToken);
         }
+
+        /// <summary>
+        /// Authenticate user session with a token.
+        /// </summary>
+        /// <param name="options"> Parameters used to authenticate user with token.</param>
+        /// <param name="cancellationToken">An optional token to cancel the request.</param>
+        /// <returns> A User and Session record.</returns>
+        public async Task<(User, Session)> AuthenticateUserWithToken(
```

Comment:
We're calling this `AuthenticateUserWithCode` now, as it is the authorization code being exchanged:

```suggestion
        public async Task<(User, Session)> AuthenticateUserWithCode(
```